send_public_api_events_to_ga/send_public_api_events_to_ga.py

```diff
@@ -0,0 +1,62 @@
+from datetime import datetime
+import csv
+import grequests
+import gzip
+import io
+import re
+import urllib
+
+EVENT_REGEX = re.compile(r'^ObjectCreated:')
+PUBLIC_API_LOG_REGEX = re.compile(r'^public_api_logs/')
+
+import boto3
+S3 = boto3.client('s3')
+
+def handle_lambda(event, context):
+    if event['Records']:
+        record = event['Records'][0]
+        file_created = EVENT_REGEX.match(record['eventName'])
+        filename = get_filename(record)
+        is_public_api_log_event = PUBLIC_API_LOG_REGEX.match(filename)
+        if file_created and is_public_api_log_event:
+            bucket_name = get_bucket_name(record)
+            obj = S3.get_object(Bucket=bucket_name, Key=filename)
+            send_events_to_GA(obj)
+
+def get_bucket_name(record):
+    return record['s3']['bucket']['name']
+
+def get_filename(record):
+    return urllib.parse.unquote(record['s3']['object']['key'])
+
+def calculate_time_delta(timestamp):
+    real_hit_time = datetime.fromtimestamp(int(timestamp))
+    timedelta = datetime.now() - real_hit_time
+    return int(timedelta.total_seconds() * 1000)
+
+def open_for_read(s3_object):
+  return io.TextIOWrapper(gzip.GzipFile(fileobj=s3_object['Body'], mode='r'))
+
+def send_events_to_GA(s3_object):
+    urls = []
+    csvreader = csv.reader(open_for_read(s3_object), delimiter="\t")
+    for row in csvreader:
+        timestamp, status, path, ip, referrer, user_agent, ga_client_id = row
+        timestamp = timestamp or int(round(datetime.now().timestamp()))
+        params = urllib.parse.urlencode({
+	    'v': 1,
+	    'tid': 'UA-26179049-14',
+	    'cid': ga_client_id or 'No client ID present',
+	    't': 'event',
+	    'ec': 'Public API request',
+	    'ea': path or 'No path present',
+	    'el': referrer or 'No referrer present',
+	    'ua': user_agent or 'No user agent present',
```

Comment:
For these things when we have an `or` might it be better to just not send them? Feels like we pollute the store with these incorrect values.

Reply:
Looking at the GA measurements protocol hit builder it's valid to not send param values for these so I'll remove the `or` values.